### PR TITLE
Optional output image name [REVPI-2886]

### PR DESCRIPTION
**Pull request by nbuchwitz**
Add an optional parameter for the name of the output image. If this name is given, the imagebakery will copy the source image to it and apply all modifications. Otherwise, a warning message is displayed, which the user must acknowledge before the program can continue. The modifications will be applied in-place as before in the aforementioned case.

### CODE REVIEW

**Comment by iluminat23** [comment]
What happens if dest = source?
This changes the behavior significantly and the old behavior is only available in interactive mode, which might not be desired.

**Comment by nbuchwitz** [comment]
> What happens if dest = source? This changes the behavior significantly and the old behavior is only available in interactive mode, which might not be desired.

We discussed this during refinement and decided that this kind of "regression" would be OK. To allow non-interactive usage we can add something like `-f` ?

**Comment by iluminat23** [comment]
This might be an option

**Comment by nbuchwitz** [comment]
See [ceb6436](https://github.com/RevolutionPi/imagebakery/pull/96/commits/ceb643692c25f9ef0a4b97cab8e8fbc44480aee4)

File: customize_image.sh [hunk 1]
```diff
@@ -64,12 +67,15 @@ OUTPUT_IMAGE=${2:-$1}
 
 if [ "$SOURCE_IMAGE" == "$OUTPUT_IMAGE" ]; then
 	echo "WARNING: No name for the output image was specified. The source image will be modfied in-place."
-	echo ""
-	echo -n "Do you want to continue? [Ny] "
-	read -r choice
 
-	if ! [[ "$choice" == "y" || "$choice" == "y" ]]; then
-		exit
+	if [ $FORCE == 0 ]; then
```
Comment (iluminat23) [inline customize_image.sh:71]: Shouldn't this be -eq as it is an arithmetic comparison?

Comment (nbuchwitz) [reply customize_image.sh]: You're right. Fixes this. Thanks.

File: customize_image.sh [hunk 2]
```diff
@@ -54,11 +56,31 @@ eval set -- "$MYOPTS"
...
+
+if [ "$SOURCE_IMAGE" == "$OUTPUT_IMAGE" ]; then
+	echo "WARNING: No name for the output image was specified. The source image will be modfied in-place."
+
+	if [ $FORCE -eq 0 ]; then
+		echo ""
+		echo -n "Do you want to continue? [Ny] "
+		read -r choice
+
+		if ! [[ "$choice" == "y" || "$choice" == "y" ]]; then
```
Comment (naruxde) [inline customize_image.sh:76]: Do you mean a capital Y in the "or" part? To check y/Y?

File: customize_image.sh [hunk 3]
```diff
@@ -54,11 +56,31 @@ eval set -- "$MYOPTS"
...
+			exit
+		fi
+	fi
+else
+	cp "$SOURCE_IMAGE" "$OUTPUT_IMAGE"
```
Comment (naruxde) [inline customize_image.sh:81]: Do somebody know a "verbose" copy command, just to show the user "We are working and alive...". This command could take a "very" long time - would me confuse :D

**Review by naruxde — CHANGES_REQUESTED** [review]
I haven't tested it yet, but will do so

File: customize_image.sh (hunk 2 above)
Comment (nbuchwitz) [reply customize_image.sh]: Thanks. Fixed in latest push

File: customize_image.sh (hunk 3 above)
Comment (nbuchwitz) [reply customize_image.sh:82]: I've added a info message before the copy starts

**Review by iluminat23 — APPROVED** [review]

**Review by naruxde — APPROVED** [review]

**Pull request merged**